doc/examples/denoise_ascm.py

```diff
@@ -0,0 +1,184 @@
+    block_radius=1,
+    rician=True)
+
+"""
+Non-local means with larger patch size which implies more smoothing, less
+sharpness
+"""
+
+den_large = non_local_means(
+    data,
+    sigma=sigma,
+    mask=mask,
+    patch_radius=2,
+    block_radius=1,
+    rician=True)
+
+"""
+Now we perform the adaptive soft coefficient matching. Empirically we set the
+adaptive parameter in ascm to be the average of the local noise variance,
+in this case the sigma itself.
+"""
+
+den_final = np.array(adaptive_soft_matching(data, den_small, den_large,
+                                            sigma[0]))
+
+print("total time", time() - t)
+
+"""
+Plot the axial slice of the data, it's denoised output and the residual
+"""
+
+axial_middle = data.shape[2] / 2
+
+original = data[:, :, axial_middle].T
+final_output = den_final[:, :, axial_middle].T
+difference = np.abs(final_output.astype('f8') - original.astype('f8'))
+difference[~mask[:, :, axial_middle].T] = 0
+
+fig, ax = plt.subplots(1, 3)
+ax[0].imshow(original, cmap='gray', origin='lower')
+ax[0].set_title('Original')
+ax[1].imshow(final_output, cmap='gray', origin='lower')
+ax[1].set_title('ASCM output')
+ax[2].imshow(difference, cmap='gray', origin='lower')
+ax[2].set_title('Residual')
+for i in range(3):
+    ax[i].set_axis_off()
+
+plt.savefig('denoised_ascm.png', bbox_inches='tight')
+
+print("The ascm result saved in denoised_ascm.png")
+
+"""
+.. figure:: denoised_ascm.png
+   :align: center
+
+   **Showing the axial slice without (left) and with (middle) ASCM denoising**.
+"""
+
+nib.save(nib.Nifti1Image(den_final, affine), 'denoised_ascm.nii.gz')
+
+print("Saving the entire denoised output in denoised_ascm.nii.gz")
+
+"""
+We would like to compare the outputs of the ``non_local_means`` (both with the
+larger as well as with the smaller patch radius) with the ASCM output.
+"""
+
+fig, ax = plt.subplots(1, 4)
+ax[0].imshow(original, cmap='gray', origin='lower')
+ax[0].set_title('Original')
+ax[1].imshow(den_small[..., axial_middle].T, cmap='gray', origin='lower',
+             interpolation='none')
+ax[1].set_title('NLMEANS small')
+ax[2].imshow(den_large[..., axial_middle].T, cmap='gray', origin='lower',
+             interpolation='none')
+ax[2].set_title('NLMEANS large')
+ax[3].imshow(final_output, cmap='gray', origin='lower', interpolation='none')
+ax[3].set_title('ASCM ')
+for i in range(4):
+    ax[i].set_axis_off()
+
+plt.savefig('ascm_comparison.png', bbox_inches='tight')
+
+print("The comparison result saved in ascm_comparison.png")
+
+"""
+.. figure:: ascm_comparison.png
+   :align: center
+
+   **Comparing outputs of the NLMEANS and ASCM**.
+"""
+
+"""
+In this tutorial we saw the use of adaptive soft coefficient matching (ASCM) 
+and how the corresponding DIPY function ``adaptive_soft_matching``. We observe
+that combining the information of two pre-denoised versions of the raw data,
+ASCM outperforms standard non-local means in supressing noise and preserving
+feature sharpness
```

Comment:
replace this paragraph with: 'For the above figure, we can observe that the information of two pre-denoised versions of the raw data, ASCM outperforms standard non-local means in supressing noise and preserving feature sharpness.'

This in my opinion is enough for the last paragraph of the example, and will make the example easier to read.